File: .history/ollama_only_20241209183029.py
from typing import List
from ollama import chat
from pydantic import BaseModel

class City(BaseModel):
    name: str
    country: str

class Cities(BaseModel):
    cities: List[City]

response = chat(
    model="llama3.2",
    messages=[
        {
            "role": "user",
            "content": "List 5 cities from around the world and their countries",
        }
    ],
    format=Cities.model_json_schema(),
)

city_list = Cities.model_validate_json(response.message.content)

print(city_list)

File: .history/ollama_only_20241209183029.js
import ollama from 'ollama';
import { z } from 'zod';
import { zodToJsonSchema } from 'zod-to-json-schema';

const City = z.object({
    name: z.string(),
    country: z.string(),
});

const Cities = z.object({
    cities: z.array(City)
})

const response = await ollama.chat({
    model: 'llama3.2',
    messages: [{ role: 'user', content: 'List 5 cities from around the world and their countries' }],
    format: zodToJsonSchema(Cities),
});

const cities = Cities.parse(JSON.parse(response.message.content));

console.log(cities);